### ISSUE TITLE
Does it take 7 hours to inference using the V100?

### ISSUE THREAD

**Issue by JamesTensor** [kohya-ss/musubi-tuner#205]
Thank you for the epic update! I'm using a V100 32G to infer the Wan2.1 14B 720B model, and it can generate videos normally. However, it takes 7 hours to generate one video. Is it normal to take such a long time? Or have I missed any acceleration strategies?

The inference parameters are as follows:
python mutil_wan_generate_video.py --task i2v-14B --vae_dtype fp16 --output_type video --attn_mode sdpa --ckpt_dir /root/pretrained_model/ckpts/Wan2.1-I2V-14B-720P --dit /root/pretrained_model/ckpts/Wan2.1-I2V-14B-720P/split_files/diffusion_models/wan2.1_i2v_720p_14B_fp16.safetensors --vae /root/pretrained_model/ckpts/Wan2.1-I2V-14B-720P/Wan2.1_VAE.pth --t5 /root/pretrained_model/ckpts/Wan2.1-I2V-14B-720P/models_t5_umt5-xxl-enc-bf16.pth --clip /root/pretrained_model/ckpts/Wan2.1-I2V-14B-720P/models_clip_open-clip-xlm-roberta-large-vit-huge-14.pth --flow_shift 5.0 --guidance_scale 5.0 \
        --vae_cache_cpu --blocks_to_swap 23 \
        --image_path /root/photo/huaxue/8_mid_image_high.png \
        --video_size 1280 720 \
        --video_length 81 \
        --infer_steps 50 \
        --prompt_file "/root/prompt/huaxue_4.json" \
        --save_path /root/generated_video/huaxue_gpu7_0402 \
        --seed $seed \
        --lora_multiplier 1.0 \
        --lora_weight /root/lora_train/huaxue_2/huaxue_2-000095.safetensors

**Comment by Sarania** [comment]
No that's not normal but you are using far from optimized settings(also are you running as root for some reason?). That said, I'm not sure what you can get away with on Tesla because I've never run one. I don't think you can use sageattention, but if you can try --attn_mode flash2 (or flash, xformers)(make sure to have them installed in your venv) for improved speed and memory usage. I would also recommend running with --fp8 and --fp8_scaled, you have 32GB VRAM which is good but you are trying to generate at the full 720p which takes a lot of VRAM. With fp8_scaled quality is 95%+ of unquantized but model is 60% size. Then don't swap as many blocks, you wanna aim to hit about 90-95% VRAM usage for maximum speed. I would also recommend maybe going for a lower res, it's still really good quality at around 544p resolutions but significantly lighter.

On my 4070 TS 16GB and specifically for Wan I2V, 1280x720@81 in fp8_scaled with sageattention takes about 1 hour and 20 minutes. The same but 960x544 takes about 35 minutes. Your V100 is about 2-3 times slower in absolute raw perf, but you also can't use sage which hurts a lot. Hopefully you can use at least flash2. Ultimately video gen is HEAVY though, if you want to optimize even more then https://github.com/kijai/ComfyUI-WanVideoWrapper has even more options like TeaCache(basically auto step skip if step won't cause much change anyway) and stuff but ofc requires ComfyUI.

**Comment by Sarania** [comment]
Oh one more thing:

Your card is twice as fast in fp16 as fp32. There is a feature in the pytorch nightlies that will definitely help your speed. It's not supported in official musubi but my fork has it:

```
https://github.com/Sarania/musubi-tuner/tree/add-features
```

--fp16_accumulation is the setting you want. This enables exactly what it says which costs you very little quality but speeds up Wan a lot, especially on a card like yours. But you have to install the pytorch nightly version or it's not available yet!

**Comment by FurkanGozukara** [comment]
@Sarania perhaps @kohya-ss can add it ?

i use this and very stable

torch==2.7.0.dev20250311

**Comment by Sarania** [comment]
> I'm a bit worried about including the dev version in the requirements.
> 
> I don't know if fp8 works on the V100, but if it does, adding the `--fp8` and `--fp8_scaled` options and reducing `--blocks_to_swap` as much as possible (as long as enough VRAM) may make it faster.

I agree, I recommend the fp16 accumulation occasionally but I think it's best to wait until it hits mainstream before bumping the reqs. Development is development for a reason. If users want to experiment for more speed, upgrading to a Pytorch Nightly is not a big ask and then if something else breaks they will know it's likely because it's nightly.

**Comment by Sarania** [comment]
Oh I also forgot to mention but Pytorch dev has broken torch.compile for me on HunyuanVideo! No errors just black outputs. Wan it still works fine. Both work on stable 2.6.0. But that's the kind of sneaky things that crop up in dev some times. The main code paths tend to be solid, but more advanced ones can have subtle issues.

**Comment by lllllinux** [comment]
> Oh one more thing:
> 
> Your card is twice as fast in fp16 as fp32. There is a feature in the pytorch nightlies that will definitely help your speed pretty much for free. It's not supported in official musubi yet but my fork has it(WanVideoWrapper as well, which I ported it from):

hey,did you encounter nan result or pure black output when enbale torch.backends.cuda.matmul.allow_fp16_accumulation on hunyuanvideo, i enable the fp16 accumulation based on diffusers hunyuanvideo framepack pipeline,but got  pure black result

**Comment by Sarania** [comment]
> > Oh one more thing:
> > Your card is twice as fast in fp16 as fp32. There is a feature in the pytorch nightlies that will definitely help your speed pretty much for free. It's not supported in official musubi yet but my fork has it(WanVideoWrapper as well, which I ported it from):
> 
> hey,did you encounter nan result or pure black output when enbale torch.backends.cuda.matmul.allow_fp16_accumulation on hunyuanvideo, i enable the fp16 accumulation based on diffusers hunyuanvideo framepack pipeline,but got pure black result

Yes, but only when using torch.compile with mode = "max-autotune-no-cudagraphs", when I reverted to "default" that issue went away for me! Default isn't noticeably slower on my system either.

**Comment by lllllinux** [comment]
> Yes, but only when using torch.compile with mode = "max-autotune-no-cudagraphs", when I reverted to "default" that issue went away for me! Default isn't noticeably slower on my system either.

that's weird. i'got nan result with no compile when using diffusers pipeline
this is my script:
```python
import torch
from PIL import Image
import numpy as np
from diffusers import AutoModel, WanImageToVideoPipeline
from diffusers.quantizers import PipelineQuantizationConfig
from diffusers.hooks.group_offloading import apply_group_offloading
from diffusers.utils import export_to_video, load_image
from transformers import UMT5EncoderModel
root_path = "Wan-AI/Wan2.1-I2V-14B-720P-Diffusers"
text_encoder = UMT5EncoderModel.from_pretrained(root_path, subfolder="text_encoder", torch_dtype=torch.float16)
vae = AutoModel.from_pretrained(root_path, subfolder="vae", torch_dtype=torch.float32)
transformer = AutoModel.from_pretrained(root_path, subfolder="transformer", torch_dtype=torch.float16)
torch.backends.cuda.matmul.allow_fp16_accumulation = True
# group-offloading
onload_device = torch.device("cuda")
offload_device = torch.device("cpu")
apply_group_offloading(text_encoder,
    onload_device=onload_device,
    offload_device=offload_device,
    offload_type="block_level",
    num_blocks_per_group=1
)
transformer.enable_group_offload(
    onload_device=onload_device,
    offload_device=offload_device,
    offload_type="leaf_level",
    use_stream=True
)

pipeline = WanImageToVideoPipeline.from_pretrained(
    root_path,
    vae=vae,
    transformer=transformer,
    text_encoder=text_encoder,
    torch_dtype=torch.float16
)
# pipeline.to("cuda")
pipeline.vae.to('cuda')
pipeline.text_encoder.to('cuda')
pipeline.image_encoder.to('cuda')
pipeline.vae.enable_tiling()

prompt = """
The camera rushes from far to near in a low-angle shot, 
revealing a white ferret on a log. It plays, leaps into the water, and emerges, as the camera zooms in 
for a close-up. Water splashes berry bushes nearby, while moss, snow, and leaves blanket the ground. 
Birch trees and a light blue sky frame the scene, with ferns in the foreground. Side lighting casts dynamic 
shadows and warm highlights. Medium composition, front view, low angle, with depth of field.
"""
negative_prompt = """
Bright tones, overexposed, static, blurred details, subtitles, style, works, paintings, images, static, overall gray, worst quality, 
low quality, JPEG compression residue, ugly, incomplete, extra fingers, poorly drawn hands, poorly drawn faces, deformed, disfigured, 
misshapen limbs, fused fingers, still picture, messy background, three legs, many people in the background, walking backwards
"""

output = pipeline(
    image = Image.open("125207055HBJ-D1(1).jpg"),
    prompt=prompt,
    height=720,
    width=1280,
    negative_prompt=negative_prompt,
    num_frames=81,
    guidance_scale=5.0,
).frames[0]
export_to_video(output, "output.mp4", fps=16)
```

**Comment by lllllinux** [comment]
> > > Oh one more thing:
> > > Your card is twice as fast in fp16 as fp32. There is a feature in the pytorch nightlies that will definitely help your speed pretty much for free. It's not supported in official musubi yet but my fork has it(WanVideoWrapper as well, which I ported it from):
> > 
> > 
> > hey,did you encounter nan result or pure black output when enbale torch.backends.cuda.matmul.allow_fp16_accumulation on hunyuanvideo, i enable the fp16 accumulation based on diffusers hunyuanvideo framepack pipeline,but got pure black result
> 
> Yes, but only when using torch.compile with mode = "max-autotune-no-cudagraphs", when I reverted to "default" that issue went away for me! Default isn't noticeably slower on my system either.

which compute dtype do you use? completely torch.float16?
